source/include/cfg/cs_HardwareVersions.h

```diff
@@ -91,3 +102,18 @@ static inline const char* get_hardware_version() {
 		}
 	}
 }
+
+// The string is always 11 chars, add 1 byte for the null terminator.
+static char versionString[12];
+
+static inline const char* get_hardware_version_from_uicr(const cs_uicr_data_t* uicrData) {
+	// Since you can't specify a max width, just use modulo to limit it.
```

Comment:
Comment the function itself instead of putting inline comments and describe limitations important to the person using this function.